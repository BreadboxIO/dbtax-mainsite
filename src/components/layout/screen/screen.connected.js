import { withRouter } from 'react-router-dom';
import { Screen } from './screen';

export const ScreenConnected = withRouter(Screen);
