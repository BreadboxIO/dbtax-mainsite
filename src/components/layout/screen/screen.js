import React from 'react';

import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Helmet } from 'react-helmet';
import { COMPANY_NAME } from 'constants/company-name';
import { TWITTER_HANDLE } from 'constants/twitter-handle';
import { PUBLIC_URL } from 'constants/public-url';
import styles from './screen.style.scss';
import OpenGraphImage from '../../../images/open-graph-image.png';

export const Screen = (props) => {
  const { children, className: classNameProp, metaTitle, metaDescription, url } = props;

  const className = classNames('animated', 'fadeIn', styles.root, classNameProp);
  const title = `${metaTitle ? `${metaTitle} | ` : ''}${COMPANY_NAME}`;

  return (
    <div className={className}>
      <Helmet>
        <title>{title}</title>
        {metaDescription && (
          <>
            <meta name="description" content={metaDescription} />
            <meta name="og:description" content={metaDescription} />
            <meta name="twitter:description" content={metaDescription} />
          </>
        )}
        {url && <meta name="og:url" content={PUBLIC_URL} />}
        <meta name="og:image" content={`${PUBLIC_URL}${OpenGraphImage}`} />
        <meta name="og:title" content={title} />
        <meta name="og:type" content="website" />
        <meta name="twitter:card" content="summary" />
        <meta name="twitter:creator" content={TWITTER_HANDLE} />
        <meta name="twitter:site" content={TWITTER_HANDLE} />
        <meta name="twitter:title" content={title} />
      </Helmet>
      {children}
    </div>
  );
};

Screen.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  className: PropTypes.string,
  metaDescription: PropTypes.string,
  metaTitle: PropTypes.string,
  url: PropTypes.string,
};

Screen.defaultProps = {
  children: null,
  className: '',
  metaDescription: '',
  metaTitle: '',
  url: '',
};
