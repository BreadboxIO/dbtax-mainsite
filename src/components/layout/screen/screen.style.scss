@import '../../../styles/config/variables';

.root {
  animation-duration: $duration-base * 5 !important;
}
