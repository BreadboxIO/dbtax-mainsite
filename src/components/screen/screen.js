import React, { Component } from 'react';

import { Footer } from 'components/main/footer/footer';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './screen.style.scss';

export class Screen extends Component {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
    className: PropTypes.string
  };

  static defaultProps = {
    children: null,
    className: ''
  };

  getClassNames() {
    const { className } = this.props;
    const result = classNames('animated', 'fadeIn', className, styles.root);

    return result;
  }

  render() {
    const { children } = this.props;
    const className = this.getClassNames();

    return (
      <div className={className}>
        {children}
        <Footer />
      </div>
    );
  }
}
