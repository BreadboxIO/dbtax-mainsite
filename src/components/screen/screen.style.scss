@import '../../styles/config/variables';
@import '../../styles/mixins/transition';

.root {
  @include transition(background-color);

  -webkit-overflow-scrolling: touch;
  box-sizing: border-box;
  flex: 1;
  overflow-y: auto;
  position: relative;
}
