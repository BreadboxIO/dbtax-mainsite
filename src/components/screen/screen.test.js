/* global React, create */
import { StaticRouter } from 'react-router-dom';
import { Screen } from './screen';

describe('<Screen />', () => {
  it('should render correctly', () => {
    const component = create(
      <StaticRouter>
        <Screen />
      </StaticRouter>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
