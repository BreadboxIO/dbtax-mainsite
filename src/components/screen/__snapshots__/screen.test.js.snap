// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> should render correctly 1`] = `
<div
  className="animated fadeIn root"
>
  <div
    className="root"
  >
    <div
      className="content"
    >
      <div
        className="root"
      >
        <div
          className="content"
        >
          Whether you're facing retirement or looking to better understand investments, we can help you with money questions.
        </div>
        <div
          className="action"
        >
          <button
            className="root button"
            onClick={null}
            type="button"
          >
            Get Started Today
          </button>
        </div>
      </div>
      <div
        className="root"
      >
        <div
          className="contact"
        >
          <h3>
            Contact
          </h3>
          <div
            className="content"
          >
            <div>
              <a
                href="mailto:+414-764-7330"
              >
                414-764-7330
              </a>
            </div>
            <div>
              <a
                href="https://www.google.com/maps/place/1125+Milwaukee+Ave,+South+Milwaukee,+WI+53172"
                rel="noopener noreferrer"
                target="_blank"
              >
                1125 Milwaukee Ave.
                <br />
                South Milwaukee
                , 
                WI
                 · 
                53172
              </a>
            </div>
            <div>
              <a
                href="mailto:social@dbtaxinc.com"
              >
                social@dbtaxinc.com
              </a>
            </div>
          </div>
        </div>
        <div
          className="links"
        >
          <h3>
            Links
          </h3>
          <div
            className="content"
          >
            <div>
              <a
                href="/"
                onClick={[Function]}
              >
                About
              </a>
            </div>
            <div>
              <a
                href="/"
                onClick={[Function]}
              >
                Services
              </a>
            </div>
            <div>
              <a
                href="/"
                onClick={[Function]}
              >
                Client Resources
              </a>
            </div>
            <div>
              <a
                href="/"
                onClick={[Function]}
              >
                Legal
              </a>
            </div>
            <div>
              <a
                href="/"
                onClick={[Function]}
              >
                Account Access
              </a>
            </div>
          </div>
        </div>
        <div
          className="learnMore"
        >
          <h3>
            Want to Know More?
          </h3>
          <div>
            We'll reach out via email to set up a call.
          </div>
          <div>
            <div
              className="root email"
            >
              <input
                placeholder="enter your email address..."
                type="text"
                value=""
              />
              <button
                className="root root"
                onClick={null}
                type="button"
              >
                <i
                  className="fas fa-paper-plane"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="root"
      >
        <div
          className="logoContainer"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="DB Tax & Financial Services"
              className="logo"
              src="db-tax-logo-dark.png"
            />
          </a>
        </div>
        <div
          className="copyright"
        >
          © 2019 DB Tax & Financial Services, Inc. All Rights Reserved.
        </div>
        <div
          className="social"
        >
          <button
            className="root root"
            onClick={null}
            type="button"
          >
            <i
              className="fab fa-facebook-f"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
